### PR TITLE
Manual cherrypick of #65034 to 1.8: make json serializer case sensitive

### PR DESCRIPTION
**Pull request by nikhita**
fixes partially https://github.com/kubernetes/kubernetes/issues/64612

This PR imports the latest jsoniterator library so that case sensitivity during unmarshalling is optional. The PR also sets Kubernetes json serializer to be case sensitive.

**Release note**:

```release-note
ACTION REQUIRED: Kubernetes JSON deserializer is now case-sensitive to restore compatibility with pre-1.8 servers. If your config files contains fields with wrong case, the config files will be now invalid.
```

/sig api-machinery
/kind bug
/assign caesarxuchao liggitt thockin sttts mbohlool

### CODE REVIEW

**Comment by nikhita** [comment]
@jpbetz for some context:

- https://github.com/kubernetes/kubernetes/issues/64612 is a bug which reported that our json serializers were case-insensitive from 1.8 onwards. This was due to a bug in https://github.com/json-iterator/go and was fixed [upstream](https://github.com/json-iterator/go/pull/282) -- so we are bumping it in this PR.
- https://github.com/kubernetes/kubernetes/pull/65034 is the original fix for 1.11, making them case-sensitive. 
- https://github.com/kubernetes/kubernetes/pull/65157 is the cherry-pick to 1.10.
- https://github.com/kubernetes/kubernetes/pull/65298 is the cherry-pick to 1.9 and contains some [discussion](https://github.com/kubernetes/kubernetes/pull/65298#issuecomment-399214523) on why this should be cherry-picked even if it is a breaking change on a patch release.
- This is the cherry-pick for 1.8.

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go [hunk 1]
```diff
@@ -98,6 +98,35 @@ func init() {
...
+		SortMapKeys:            true,
+		ValidateJsonRawMessage: true,
+		CaseSensitive:          true,
+	}.Froze()
+}
+
+var caseSensitiveJsonIterator = CaseSensitiveJsonIterator()
+
+// gvkWithDefaults returns group kind and version defaulting from provided default
+func gvkWithDefaults(actual, defaultGVK schema.GroupVersionKind) schema.GroupVersionKind {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go]: Is this needed?

**Review by caesarxuchao — COMMENTED** [review]
One nit, other wise lgtm

**Comment by caesarxuchao** [comment]
#65419 needs to also be merged before making a new release.

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go (hunk 1 above)
Comment (nikhita) [reply staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go]: Removed. :+1:

**Comment by nikhita** [comment]
/hold
for fix to scheduler config decoding (#65424, 1.8 pick in #65419)

**Comment by nikhita** [comment]
scheduler config decoding has been fixed and cherry-pick to 1.8 has been merged: https://github.com/kubernetes/kubernetes/pull/65419.

Removing hold.
/hold cancel

**Comment by nikhita** [comment]
@thockin @lavalamp can either of you approve? thanks!

**Comment by jpbetz** [comment]
I've approved this as patch manager since I'm familiar with the change and it's already going in to the other branches.

**Pull request merged**